### DISCUSSION TITLE
Best protocol and encoding for rtl_433

### DISCUSSION THREAD

**Discussion by hakelm** [merbanan/rtl_433#2917] [General]
I would like to set up a small battery powered 433 MHz transmitter for the transmission to a rtl dongle of sea level data where wifi and cable are impractical. The data is a simple 3-digit number and will be transmitted once an hour. The data is collected and delivered to a simple transmitter by an 8-bit PIC microcontroller.
Can someone in this group give me some advice on selection of protocol, pulse widths etc or in the best of worlds some ready made software to use.
Thanks in advance for any tip.
H

**Comment by zuckschwerdt** [comment]
Did you already choose a transmitter for your project? I would recommend a capable tranceiver like the CC1101 or any of the similar FSK-capable chips. Simply connect it by I2C, then upload RF specs with configuration commands and offload sending by uploading a data buffer. An optimal protocol (preamble, sync-word, data, checksum with e.g. MC coding) is already handled by the tranceiver.
Writing a custom decoder for rtl_433 is simple enough, but you could also emulate some existing device and extract data from a fake temperature/humidity reading or such.

Or you could build a bridge with a CC1101 or alike on both sides, or maybe even use a proprietary RF protocol with something like a nRF24L01 pair (look for modules with external antenna, PA, LNA).

**Reply by hakelm** [reply]
Thanks, a couple of years ago I had a project with nrf241 that I abandoned due to an unresolved problem
(https://devzone.nordicsemi.com/f/nordic-q-a/72523/nrf24l01-reset). I still have a dozen or so of the devices and also the software.  I will look into the the nrf241 again.
Otherwise the CC1101 may be the better alternative.
H
